### PR TITLE
Auth in node

### PR DESCRIPTION
**Pull request by JasonSowers**

### CODE REVIEW

File: javascript_nodejs/19.bot-authentication/bot.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
+const { ActivityTypes, MessageFactory } = require('botbuilder');
+const { ChoicePrompt, DialogSet, WaterfallDialog, OAuthPrompt } = require('botbuilder-dialogs');
+const CONNECTION_NAME = "";
```
Comment (stevengum) [inline javascript_nodejs/19.bot-authentication/bot.js]: This should have a comment explaining what it is and what are acceptable values.

File: javascript_nodejs/19.bot-authentication/bot.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+const AUTH_DIALOG = 'auth_dialog';
+const HELP_TEXT = ' Type anything to get logged in. Type \'logout\' to signout.' +
+                  ' Type \'help\' to view this message again';
+const OAUTH_SETTINGS = {
+    connectionName: CONNECTION_NAME,
+    title: "Sign In",
+    text: "Please Sign In",
+    timeout: 300000
+};
+// Creat an OAuthPrompt
```
Comment (stevengum) [inline javascript_nodejs/19.bot-authentication/bot.js]: Typos, punctuation

File: javascript_nodejs/19.bot-authentication/bot.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        this.conversationState = conversationState;
+        this.dialogState = this.conversationState.createProperty(DIALOG_STATE_PROPERTY);
+        this.dialogs = new DialogSet(this.dialogState);
+
+        // Add prompts that will be used by the bot.
+
+        this.dialogs.add(new ChoicePrompt(CONFIRM_PROMPT));
+        this.dialogs.add(new OAuthPrompt(OAUTH_PROMPT, OAUTH_SETTINGS))
+
+        // the waterfall dialog that controls the flow of the conversation
```
Comment (stevengum) [inline javascript_nodejs/19.bot-authentication/bot.js]: Capitalization and punctuation

File: javascript_nodejs/19.bot-authentication/README.MD [hunk 4]
```diff
@@ -0,0 +1,47 @@
+[![Deploy to Azure](http://azuredeploy.net/deploybutton.png)](https://azuredeploy.net/)
+
+This sample shows how to use authentication in your bot using OAuth. This bot example uses [`restify`](https://www.npmjs.com/package/restify) and [`dotenv`](https://www.npmjs.com/package/dotenv).
+
+# To try this sample
+- Clone the repository
+    ```bash
+    git clone https://github.com/Microsoft/botbuilder-sample.git
```
Comment (stevengum) [inline javascript_nodejs/19.bot-authentication/README.MD]: botbuilder-samples

File: javascript_nodejs/19.bot-authentication/bot.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+
```
Comment (stevengum) [inline javascript_nodejs/19.bot-authentication/bot.js]: Extra blank line here

File: javascript_nodejs/19.bot-authentication/bot.js (hunk 2 above)
Comment (JasonSowers) [reply javascript_nodejs/19.bot-authentication/bot.js]: don't worry we will find it a home 👍

File: javascript_nodejs/19.bot-authentication/bot.js [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+        // Add prompts that will be used by the bot.
+
+        this.dialogs.add(new ChoicePrompt(CONFIRM_PROMPT));
+        this.dialogs.add(new OAuthPrompt(OAUTH_PROMPT, OAUTH_SETTINGS))
+
+        // The waterfall dialog that controls the flow of the conversation.
+        this.dialogs.add(new WaterfallDialog(AUTH_DIALOG, [
+            async(dc) => {
+                return await dc.prompt(OAUTH_PROMPT);
```
[the remark this answers is not in the document]
Comment (JasonSowers) [reply javascript_nodejs/19.bot-authentication/bot.js]: I'm ready to fix this, but I am waiting on a change to the node sdk @stevengum and @benbrown were telling me about.

**Comment by benbrown** [comment]
I updated this sample to remove BotStateSet and to use named methods instead of lambdas.

**Pull request merged**